manager/orchestrator/taskreaper/task_reaper.go

```diff
@@ -282,6 +307,39 @@ func (tr *TaskReaper) tick() {
 			return nil
 		})
 	}
+
+	// Perform service cleanup
+	deleteServices := make(map[string]struct{})
+	// figure out which services need to be cleaned up
+	if len(tr.cleanupServices) > 0 {
+		tr.store.View(func(tx store.ReadTx) {
+			for svc := range tr.cleanupServices {
+				tasks, err := store.FindTasks(tx, store.ByServiceID(svc))
+				// if all tasks of svc have been cleaned up already, then we should get a response
+				// with length zero. If an error occurs though, we log it and don't mark the service
+				// for removal yet.
+				if err != nil {
+					log.G(context.Background()).WithError(err).Errorf("error in task reaper while looking up tasks for service %s", svc)
+					continue
+				}
+				if len(tasks) == 0 {
```

Comment:
len(tasks) may not be zero. Consider the case of deleting a service whose tasks were already shutdown. Another case is when you restart the reaper and there are services from previous term which need to be deleted but may have services to be removed. You'll need to check the desired and actual state of each task if the len(tasks) > 0

Reply:
I agree, this will need to be fixed.

Reply:
@anshulpundir I've added a fix for this. This required another change that tasks of the service that was being removed needed to be newly tracked for removal before the tick was finished. PTAL.